source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,156 @@
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding only a subset of them in memory at a
+given time. Read operations that match multiple documents use a cursor
+to return those documents in batches as opposed to all at once.
+
+Sample Cursor
+~~~~~~~~~~~~~
+
+Each section uses the following ``cursor`` variable, which is a
+``Cursor`` struct that contains all the documents in a collection:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+.. important::
+
+   A cursor is not goroutine safe. Do not use the same cursor in
+   multiple goroutines at the same time.
+
+.. _individual-documents-golang:
+
+Retrieve Documents Individually
+-------------------------------
+
+To retrieve documents from your cursor individually, use the
+``Next()`` method.
+
+This method blocks the current goroutine. That means the function
+returns a document if each of the following is true:
+
+- A document is currently or will later be available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. include:: /includes/fundamentals/exhausted-cursor-definition.rst
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor next
+   :end-before: end cursor next
+
+Documents Only in the Cursor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To attempt retrieving a document from your cursor, use the ``TryNext()``
+method.
+
+This method is unable to block the current goroutine. That means the
+function returns a document if each of the following is true:
+
+- A document is currently available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. include:: /includes/fundamentals/exhausted-cursor-definition.rst
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor try next
+   :end-before: end cursor try next
+
+Retrieve All Documents
+----------------------
+
+To populate an array with all of your query results, use the ``All()``
+method:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor all
+   :end-before: end cursor all
+
+.. important:: Memory
+
+   If the number and size of documents returned by your query exceeds
+   available application memory, your program will crash. If you except
+   a large result set, you should :ref:`consume the cursor iteratively
+   <individual-documents-golang>`.
+
+Close the Cursor
+----------------
+
+When your application no longer needs to use a cursor, close the cursor
+with the ``Close()`` method. This method frees up your cursor's
+consumption of resources in both the client application and the 
```

Comment:
Suggestion: I think "frees up" is idiomatic. Consider deleting "up".
Suggestion: I think "frees consumption..." should be "frees the resources" (I do not think "consumption" can be freed). 

```suggestion
with the ``Close()`` method. This method frees the resources your cursor consumes in both the client application and the 
```